server/service/topicService.js

```diff
@@ -1035,5 +1036,60 @@ exports.getRecentTopicEnrollmentEvents = async function(limit) {
     catch(e) {
         console.log(e.stack);
         return false;
-    }  
+    }
+}
+
+/**
+ * Get all topics that are visible to a user. This includes topics that are owned by the user and active,
+ * topics that are and shared with the user, and topics that are publicly available.
+ * TODO: Currently this method is implementing user owned active and publicly available topics. This will be updated to include shared topics
+ * when the shared topics model is implemented.
+ * @param {int} ownerId
+ * @param {int} limit Optional - If provided, will return up to the limit number of topics if not provided will return up to 100 topics
+ * @param {int} offset Optional - If provided, will return topics starting at the offset otherwise will start at the beginning
+ * @returns All visible topics as an array
+ */
+exports.getAllVisibleTopics = async ( ownerId , limit, offset) => {
+
+    if( ownerId > -1 ) {
+
+        let text = "select * from topics WHERE active = $1 and (owned_by = $2 OR visibility = 2) ORDER BY id";
```

Comment:
Great work on this fix. I replied in Discord before seeing this pull request, but glad to see the correction. Well done.